lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -484,6 +484,9 @@ LargeHeapBlock::AllocFreeListEntry(size_t size, ObjectInfoBits attributes, Large
 
     header->objectIndex = headerIndex;
     header->objectSize = originalSize;
+#ifdef RECYCLER_WRITE_BARRIER
+    header->hasWriteBarrier = (attributes&WithBarrierBit) == WithBarrierBit;
```

Comment:
this is told by the allocator, passing in the WithBarrierBit. currently the mix mode won't happen because we either turn on all with writebarrier or, only turn on for function body, which is not in large block. in the future if we selectively enable more write barrier allocation on windows, we will hit such scenario. and I had a comment in the code to use dedicated page allocator for large block with barrier.
fixing the spaces

---
In reply to: [88134866](https://github.com/Microsoft/ChakraCore/pull/1968#discussion_r88134866) [](ancestors = 88134866)